### PR TITLE
Add dtype -1 condition in aten_scalar_tensor | feat(torch_lib)

### PR DESCRIPTION
**Pull request by titaiwangms**

### CODE REVIEW

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 1]
```diff
@@ -5141,11 +5141,13 @@ def aten_rsub(self: TReal, other: TReal, alpha: float = 1.0) -> TReal:
...
-def aten_scalar_tensor(s: float, dtype: int = FLOAT.dtype) -> TTensor:  # type: ignore[type-var]
+@torch_op("aten::scalar_tensor", trace_only=True)
+def aten_scalar_tensor(s: float, dtype: int = -1) -> TTensor:  # type: ignore[type-var]
     """scalar_tensor(Scalar s, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor"""
-
-    return op.Cast(s, to=dtype)
+    # If dtype is not specified in torch, aten would put -1 as default.
+    if dtype != -1:
+        return op.Cast(s, to=dtype)
+    return op.Cast(s, to=FLOAT.dtype)
```
Comment (titaiwangms) [inline onnxscript/function_libs/torch_aten/ops/core.py:5150]: Following original behavior of this function. However, I am not sure what we should do with -1. Is CastLike itself better?

Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:5150]: Why is this change needed?

Comment (titaiwangms) [reply onnxscript/function_libs/torch_aten/ops/core.py:5150]: Because when dtype is None, aten doesn't give us None, they give us -1

**Comment by titaiwangms** [comment]
Closed as we should disable dtype -1 default assignment in exporter instead.

**Comment by justinchuby** [comment]
For future reference, we need to strive to preserve existing scripted functions and not make them trace_only when possible. trace_only should be seen as a compromise and not preferred.

**Pull request closed without merging**